package/src/components/Message/MessageSimple/utils/renderText.tsx

```diff
@@ -242,10 +246,54 @@ export const renderText = <
     );
   };
 
+  const listLevels = {
+    sub: 'sub',
+    top: 'top',
+  };
+
+  /**
+   * For lists and sublists, the default behavior of the markdown library we use is
+   * to always renumber any list, so all ordered lists start from 1.
+   *
+   * This custom rule overrides this behavior both for top level lists and sublists,
+   * in order to start the numbering from the number of the first list item provided.
+   * */
+  const customListAtLevel =
```

Comment:
Naming curried functions is a bit tricky in general, since this technically gets called like `customListAtLevel('top')(node, output, state)`. My thinking here is that if you start looking at the list of custom rules, it looks like `customListAtLevel('top')`, so I was considering it easier to read this as "I'm setting a custom rule for a list at the level top", but letting the type system deal with the types directly itself. My IDE/editor will tell me it's returning a ReactNodeOutput, and I don't necessarily care about that when trying to see what's going on in the file.

I'm open to changing it of course, but I'm afraid it's easy to lose the understanding of the fact the currying/selection of level is what's happening with the first argument. But if you have any ideas on how to go about naming it better, I'm all ears for suggestions.

Reply:
It's good to point out though, the convenience of currying it here does make it a bit trickier to figure this kind of thing out. I love functional programming, but the half measure that often exists in js/ts can be tricky :D